### DISCUSSION TITLE
Webpack module federatiom consuming remotes

### DISCUSSION THREAD

**Discussion by crhenrici** [nuxt/nuxt#16007] [Questions]
I'm currently developing a POC Microfrontend with Nuxt 3 using Module Federation from Webpack 5, which Nuxt 3 should be supporting.

For the webpack configuration I configured it in the nuxt.config.ts, similarly to how I would do it with Nuxt 2 since there was nothing specified in the documentation.
```
import { defineNuxtConfig } from 'nuxt3'
import ModuleFederationPlugin from 'webpack/lib/container/ModuleFederationPlugin'

export default defineNuxtConfig ({
  build: {
    plugins: [
      new ModuleFederationPlugin({
        name: 'test',
        filename: 'remoteEntry.js',
        remotes: {
          "second": "second@http://localhost:8080/remoteEntry.js"
        },
        exposes: {},
        shared: ["vue"]
      })
    ]
  }
})
```
I'm trying to import a Vue component which I exposed with the Module Federation plugin. The exposing should be correct as I also tested it with a plain Vue app and consumed it like described above.

I'm trying to import the component in my app like this in the `nuxt3-app/pages/view/dynamic.vue` file.
```
<template>
    <div>
        <h1>Dynamic Page</h1>
        <TestVue msg="nuxt3-app" :config="$config"/>
    </div>    
</template>
<script>
import TestVue from "second/TestVue"

export default {
    name: 'dynamic',
    data() {
        return {
             $config: {
                foo: "bar",
                modules: [{
                name: "TestVue",
                component: "TestVue",
                url: "second/TestVue"
                }]
            }
        }
    },
    components: {
        TestVue
    }
}
</script>
```
Here again I tested this approach with the other Vue app so this should work.

Yet when I run the application it's not able to import the component as I receive the following warning message 
```
[SSR] Error transforming C:/Users/CristianHenrici/Microfronted-Test/nuxt3-app/components/test.vue: Error: Failed to resolve import "second/TestVue" from "components\test.vue". Does the file exist? 
```
and eventually when routing to the page I receive the following error

```
[worker] Cannot read properties of undefined (reading '__vccOpts')                                     
  at node_modules\vue-router\dist\vue-router.cjs.js:2022:55
```
Am I missing something during the configuration or has the configuration of webpack changed for Nuxt 3. I was researching this intensively for the last few days without any success so any help, pointer etc. would be greatly appreciated.

**Comment by danielroe** [comment]
Possibly relevant: https://github.com/nuxt/framework/pull/2223

**Reply by crhenrici** [reply]
Thanks, I was able to find a "working" example there. But looks like I have to wait until this mentioned issue is fixed

**Comment by AndreyYolkin** [comment]
So, I think we really need a documentation of how to implement MF in nuxt 3, because author #2223 provided example with nuxt in origin and vue in remote, but probably we need nuxt in both applications

**Comment by fro-dev-clau** [comment]
Someone has a working example of remote/host using nuxt3?

**Comment by dkarnacewicz** [comment]
@danielroe  is there any working example or documentation how to implement MF of remote/host using nuxt3?
